### PR TITLE
wallet2: don't try to lock an empty filename

### PR DESCRIPTION
**Pull request by moneromooo-monero**

### CODE REVIEW

File: src/wallet/wallet2.cpp [hunk 1]
```diff
@@ -7577,6 +7581,8 @@ bool wallet2::unlock_keys_file()
 
 bool wallet2::is_keys_file_locked() const
 {
+  if (m_wallet_file.empty())
+    return false;
```
Comment (xiphon) [inline src/wallet/wallet2.cpp]:
```suggestion
    return true;
```

Comment (moneromooo-monero) [reply src/wallet/wallet2.cpp]: Why ?

Comment (xiphon) [reply src/wallet/wallet2.cpp]: Cause when i lock the file, i expect it to be locked.

Actually would be correct to introduce a bool storing the lock state.

Comment (moneromooo-monero) [reply src/wallet/wallet2.cpp]: OTOH, if you unlock it, you expect it to be unlocked :) Claiming locked all the time would prevent the wallet starting. Adding state would also work though.

Comment (xiphon) [reply src/wallet/wallet2.cpp]: Given that it doesn't have state right now, it has to return `true` at least.
`false` will result in throwing the following exception:

https://github.com/monero-project/monero/blob/4b38400336c2bd5b0d011da980ca9e46862a50b9/src/wallet/wallet2.cpp#L5426

Comment (moneromooo-monero) [reply src/wallet/wallet2.cpp]: How can you get this ? The patch passed the functional tests, which use memory wallets (and were trying to lock ".keys").

Comment (xiphon) [reply src/wallet/wallet2.cpp]: Not sure there is a way to trigger the condition manually or through RPC interface.

Comment (moneromooo-monero) [reply src/wallet/wallet2.cpp]: This is only called if there is a file to load from AFAICT.

**Pull request merged**